prairielib/lib/sql-db.js

```diff
@@ -419,14 +479,35 @@ module.exports.query = function(sql, params, callback) {
             return true;
         };
         if (handleError(err)) return;
-        paramsToArray(sql, params, function(err, newSql, newParams) {
-            if (err) err = error.addData(err, {sql: sql, sqlParams: params});
+
+        const setSearchPath = function(callback) {
+            if (searchSchema != null) {
+                const setSearchPathSql = `SET search_path TO ${searchSchema},public`;
```

Comment:
For legacy reasons the "explicit client" functions (`getClient()`, `queryWithClient()`, `queryWithClientOneRow()`, etc.) are completely separate from the "implicit client" functions (`query()`, `queryOneRow()`, etc.). This means that the `SET search_path` needs to be done in two separate places in the code.

Any given caller will either be using the "explicit client" functions, so they'll get a client with `search_path` set, or they just be doing a one-off `queryXXX()` call with an implicit client, which will set the `search_path` for just that single call.

Maybe it would be cleaner to rewrite `query()` to use `getClient()` underneath, so that we only have to do `SET search_path` in one place?

Reply:
I rewrote `query()` to just call `getClient()` and `queryWithClient()`. This is a pretty fundamental function to rewrite, but the new version is really short and seems safe. This is a big cleanup in that the old `query()` was pretty messy and we now only have `SET search_path` in one place in the code.